java/test_flatbuf1/Test.java

```diff
@@ -16,16 +17,246 @@
     Test() {
         /* Create an instance of the DDlog program with one worker thread. */
         this.api = new DDlogAPI(1, null, false);
-        api.record_commands("replay.dat", false);
+        api.recordCommands("replay.dat", false);
     }
 
     void check(int exit) {
         if (exit != 0)
             throw new RuntimeException("Exit code shows error" + exit);
     }
 
-    void onCommit(DDlogCommand command) {
-        System.out.println(command.toString());
+    
+    // typedef tuple = (bool, bit<8>, string)
+    private String printTuple(Tuple3__bool__bit_8___stringReader t) {
+        return "(" + t.a0() + ", " + t.a1() + ", \"" + t.a2() + "\")";
+    }
+ 
+    // typedef Many = A{x: string}
+    //              | B{b: bool}
+    //              | D{t: tuple}
+    private String printMany(ManyReader m) {
+        if (m instanceof ManyReader.A) {
+            return "A{\"" + ((ManyReader.A)m).x() + "\"}";
+        } else if (m instanceof ManyReader.B) {
+            return "B{" + ((ManyReader.B)m).b() + "}";
+        } else if (m instanceof ManyReader.D) {
+            return "D{" + printTuple(((ManyReader.D)m).t()) + "}";
+        } else {
+            throw new IllegalArgumentException("Invalid Many value " + m.toString());
+        }
+    }
+
+    void onCommit(DDlogCommand<Object> command) {
+        int relid = command.relid();
```

Comment:
One other way we can also test this is to print each record twice: once using DDlogRecord.toString(), and once using your method. Then we should check that all outputs appear twice.

Reply:
done in the new PR